Sources/Shared/UI/Social/MessageComposerViewController.m

```diff
@@ -328,65 +334,60 @@ - (IBAction)onBtnSend:(id)sender
-                
-                //release the date formatter because, not needed after that piece of code
-
-                fileAttachName = [NSString stringWithFormat:@"%@%@.png",MOBILE_UPLOAD_FILE_PREFIX, fileAttachName];
-                fileAttachName = [fileAttachName stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
-                NSLog(@"uploading file: %@",fileAttachName);
-                
-                fileAttachURL = [NSString stringWithFormat:@"%@/%@/%@",photosFolderPath,MOBILE_UPLOAD_DEST_FOLDER,fileAttachName];
-                
-                NSData *imageData = UIImagePNGRepresentation(self.attPhotoView.image);
-                
-                NSInvocation *invocation = [NSInvocation invocationWithMethodSignature:
-                                            [self methodSignatureForSelector:@selector(sendImageInBackgroundForDirectory:data:)]];
-                [invocation setTarget:self];
-                [invocation setSelector:@selector(sendImageInBackgroundForDirectory:data:)];
-                [invocation setArgument:&fileAttachURL atIndex:2];
-                [invocation setArgument:&imageData atIndex:3];
-                [NSTimer scheduledTimerWithTimeInterval:0.1f invocation:invocation repeats:NO];
-                
-            }
-        }
-        
-        
-        if(_isPostMessage)
-        {
-            [self displayHudLoader];
-            
-            self.postActivityProxy = [[SocialPostActivity alloc] init];
-            self.postActivityProxy.delegate = self;
-
-            [self.postActivityProxy postActivity:_txtvMessageComposer.text fileURL:fileAttachURL fileName:fileAttachName toSpace:selectedSpace];
-        }
-        else
-        {
-            [self displayHudLoader];
+            fileAttachName = [NSString stringWithFormat:@"%@%@.png",MOBILE_UPLOAD_FILE_PREFIX, fileAttachName];
+            fileAttachName = [fileAttachName stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
+            NSLog(@"uploading file: %@",fileAttachName);
             
-            self.postCommentProxy = [[SocialPostCommentProxy alloc] init];
-            self.postCommentProxy.delegate = self;
-            [self.postCommentProxy postComment:_txtvMessageComposer.text forActivity:_strActivityID];
+            NSData *imageData = UIImagePNGRepresentation(self.attPhotoView.image);
             
+//            2. Create the upload manager
+            [self showUploadProgressViewController];
+            uploadManager = [[MobileFileUploadManager alloc] init];
+            /*
+            3. Ask to upload the att photo with 2 handler blocks to get the upload process status & completion signal
+            4a. In case completion success: request to create the match activity
+            4b. In case of completion failed: ask to post the message any way if error connection or something. If beause use has tapped on cancel button, just close the upload view controller
```

Comment:
It's good to put comments explaining what is happening, it's better if they are not full of typos ;) : because, user